Maple2.Server.Game/Manager/Items/InventoryManager.cs

```diff
@@ -159,15 +159,16 @@ public class InventoryManager {
                 // Slot MUST be -1 so we don't add directly to a slot.
                 add.Slot = -1;
                 int remainStack = items.GetStackResult(add);
-                if (remainStack > 0) {
-                    using GameStorage.Request db = session.GameStorage.Context();
-                    Item? newAdd = db.CreateItem(session.CharacterId, add);
-                    if (newAdd == null) {
-                        return false;
-                    }
-
-                    add = newAdd;
+                if (remainStack == 0 && items.OpenSlots == 0) {
```

Comment:
This is still wrong. If `remainStack` is 0, we CAN add but do not stack. the full logic should be something like:

```cs
int remainStack = items.GetStackResult(add);
if (remainStack > 0) {
    if (items.OpenSlots <= 0) {
        return false;
    }

    using GameStorage.Request db = session.GameStorage.Context();
    Item? newAdd = db.CreateItem(session.CharacterId, add);
    if (newAdd == null) {
        return false;
    }

    add = newAdd;
}
```